### PR TITLE
fix: add default state for pullRequestCountBy and issueCountBy built-ins in transform

### LINKED ISSUE #434: pullRequestCountBy builtin requires state
**Describe the bug**
According to reviewpad documentation, the builtin `pullRequestCountBy` has two parameters. The first one is set as required but the second is set as optional.

However, if the second parameter is not provided an error is returned:

```
Error: type inference failed: mismatch in arg types on pullRequestCountBy
```

**To Reproduce**
1. Navigate to reviewpad [playground](https://reviewpad.com/playground)
2. Paste into the editor the following configuration:

```yaml
api-version: reviewpad.com/v3.x

workflows:
  - name: welcome-user
    if:
      - rule: $pullRequestCountBy($author()) == 1
    then:
      # $commentOnce() comments on the pull request once
      # For more details see https://docs.reviewpad.com/guides/built-ins/#commentonce
      - $commentOnce("Thank you for your first contribution.")

      # An alternative is to comment with the pull request author's name
      # - $commentOnce($sprintf("Thank you @%s for your first contribution!", [$author()]))
```

3. Click in the button `RUN`.
4. The following error should appear.

```
> Error: type inference failed: mismatch in arg types on pullRequestCountBy
```

**Expected behavior**
The parameter `state` should be optional and no error should happen if not provided.

### PR DESCRIPTION
**Pull request by zolamk**
## Description
This PR adds a default state(all) to the `$pullRequestCountBy` and `$issueCountBy` built-ins

## Related issue

Closes #434 

## Type of change

Bug fix (non-breaking change which fixes an issue)

## How was this tested?
Unit and manual tests

## Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works 
- [x] I have ran `task check -f` and have no issues

## Code review and merge strategy (ship/show/ask) 

- [x] Ship: this pull request can be automatically merged and does not require code review

### CODE REVIEW

**Review by ferreiratiago — APPROVED** [review]
Can you please update the documentation on [docs](https://github.com/reviewpad/docs) for both [pullRequestCountBy](https://docs.reviewpad.com/guides/built-ins/#pullrequestcountby) and [issueCountBy](https://docs.reviewpad.com/guides/built-ins/#issuecountby). Also, please review the [use cases](https://github.com/reviewpad/docs/tree/main/docs/use-cases) where both `pullRequestCountBy` and `issueCountBy` are being used with the unnecessary parameter `all`.

File: engine/transform.go [hunk 1]
```diff
@@ -43,29 +43,17 @@ func addDefaultSizeMethod(str string) string {
 
 // reviewpad-an: generated-by-co-pilot
```
Comment (ferreiratiago) [inline engine/transform.go:44]: Does this comment continue to make sense?

Comment (zolamk) [reply engine/transform.go:44]: yeah, we shouldn't need it anymore

**Comment by zolamk** [comment]
> Can you please update the documentation on [docs](https://github.com/reviewpad/docs) for both [pullRequestCountBy](https://docs.reviewpad.com/guides/built-ins/#pullrequestcountby) and [issueCountBy](https://docs.reviewpad.com/guides/built-ins/#issuecountby). Also, please review the [use cases](https://github.com/reviewpad/docs/tree/main/docs/use-cases) where both `pullRequestCountBy` and `issueCountBy` are being used with the unnecessary parameter `all`.

Done @ferreiratiago, please review https://github.com/reviewpad/docs/pull/118

**Pull request merged**